scipy/ndimage/tests/test_morphology.py

```diff
@@ -1,28 +1,39 @@
 import numpy as np
-from numpy.testing import (assert_, assert_equal, assert_array_equal,
-                           assert_array_almost_equal)
+from scipy._lib._array_api import is_cupy
+from scipy._lib._array_api import (xp_assert_equal, assert_array_almost_equal)
 import pytest
 from pytest import raises as assert_raises
 
 from scipy import ndimage
 
 from . import types
 
+from scipy.conftest import array_api_compatible
+skip_xp_backends = pytest.mark.skip_xp_backends
+pytestmark = [array_api_compatible, pytest.mark.usefixtures("skip_xp_backends"),
+              # XXX: only CuPy delegation is implemented
+              skip_xp_backends("jax.numpy", "torch", "array_api_strict"),
+]
+
 
 class TestNdimageMorphology:
 
     @pytest.mark.parametrize('dtype', types)
-    def test_distance_transform_bf01(self, dtype):
+    def test_distance_transform_bf01(self, dtype, xp):
+        dtype = getattr(xp, dtype)
+        if is_cupy(xp):
+            pytest.xfail("CuPy does not have distance_transform_bf.")
+
         # brute force (bf) distance transform
-        data = np.array([[0, 0, 0, 0, 0, 0, 0, 0, 0],
+        data = xp.asarray([[0, 0, 0, 0, 0, 0, 0, 0, 0],
                          [0, 0, 0, 0, 0, 0, 0, 0, 0],
                          [0, 0, 0, 1, 1, 1, 0, 0, 0],
                          [0, 0, 1, 1, 1, 1, 1, 0, 0],
                          [0, 0, 1, 1, 1, 1, 1, 0, 0],
                          [0, 0, 1, 1, 1, 1, 1, 0, 0],
                          [0, 0, 0, 1, 1, 1, 0, 0, 0],
                          [0, 0, 0, 0, 0, 0, 0, 0, 0],
-                         [0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype)
+                         [0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=dtype)
```

Comment:
indentation here and similar cases below in this file - why not just `xp.asarray(data)` on a new line